src/api/questions.ts

```diff
@@ -130,6 +132,18 @@ export const getQuestionById = async (
  * Returns:
  *   Promise<ApiResponse<{ questions: Question[]; totalQuestions: number }>>: The questions data on success, or an error message on failure.
  */
+
+// Define the structure of the search response
+interface SearchResponse {
```

Comment:
For consistency, can we move `SearchResponse` and `GetAllResponse` to the `Question.ts` types file?

Reply:
Also I'm just curious, but is there a reason that the `SearchResponse` doesn't have the `totalQuestions`, but the `GetAllResponse` does?